activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -117,7 +117,10 @@ def calculate(operation, column_name)
       end
 
       if has_include?(column_name)
-        construct_relation_for_association_calculations.calculate(operation, column_name)
+        relation = construct_relation_for_association_calculations
+        relation = relation.distinct if operation.to_s.downcase == "count"
```

Comment:
It's the same transformation as the first line of perform_calculations.  I could move that first line `operation = operation.to_s.downcase` into calculate, but I didn't think it was necessary. Up to you.